docs/design/1-support-task-progress.md

```diff
@@ -0,0 +1,85 @@
+---
+author: Lance Ren <lanceren@yunify.com>
+status: draft
+updated_at: 2020-02-19
+---
+
+# Proposal: Support Task Progress
+
+## Background
+
+For many of our tasks, they cost a lot while running. In case of improving
+user's experience, we do need to support return task progress state.
+The task caller may use the state for implementing the process bar and so on.
+
+## Proposal
+
+So I propose following changes:
+
+- Add a type (maybe like: progress by percent) in time-cost task,
```

Comment:
We need to give clear and definite implementation proposal here:

- Which type?
- How to add?
- How to use?
- How to handle edge cases?
- How to work with other parts in noah?
- ...

And we can put example / benchmark code in `design/{proposal_id}` for reference.